src/sentry/profiles/consumers/process/factory.py

```diff
@@ -1,17 +1,23 @@
-from typing import Mapping
+from typing import Dict, Mapping
 
 from arroyo.backends.kafka.consumer import KafkaPayload
 from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
 from arroyo.processing.strategies.commit import CommitOffsets
 from arroyo.processing.strategies.run_task import RunTask
 from arroyo.types import Commit, Message, Partition
 
+from sentry import options
 from sentry.processing.backpressure.arroyo import HealthChecker, create_backpressure_step
 from sentry.profiles.task import process_profile_task
 
 
 def process_message(message: Message[KafkaPayload]) -> None:
-    process_profile_task.s(payload=message.payload.value).apply_async()
+    msg_payload = message.payload.value
+    msg_headers = get_headers_dict(message.payload.headers)
```

Comment:
Why don't you just try to get the field you need?

Reply:
@phacops nope, `Message` headers values are `String`.

Reply:
Got it. I'd still zeroed in on the field you're looking for and not convert the whole dict.

Reply:
other examples from existing `Sentry` codebase: https://github.com/getsentry/sentry/blob/master/src/sentry/eventstream/kafka/protocol.py#L168-L169